### PR TITLE
Switch to src layout

### PR DESCRIPTION
**Pull request by leroyvn**
# Description

This PR switches sources to the so-called "src layout". See [pytest docs](https://docs.pytest.org/en/6.2.x/goodpractices.html) for the motivation. In addition to moving files to the appropriate location, I made the following changes:

- Added `__init__.py` to every test subdirectory.
- Removed the disambiguating prefixes from filenames and test function names.
- Updated test report to the new source layout.

@nollety @schunkes please check whether this impacts your workflow. The source layout change will require that you fix your PyCharm configuration if you use it. You will also have to reinstall the Eradiate python package (the simplest way to do so is to rebuild your Conda env).

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

File: tests/kernel/test_gridvolume.py [hunk 1]
```diff
@@ -2,17 +2,8 @@
...
 
-from eradiate.contexts import SpectralContext
 from eradiate.kernel.gridvolume import read_binary_grid3d, write_binary_grid3d
-from eradiate.radprops.rad_profile import US76ApproxRadProfile
-
-
-@pytest.fixture
-def test_radprops(mode_mono):
-    spectral_ctx = SpectralContext.new(wavelength=550.0)
-    return US76ApproxRadProfile().to_dataset(spectral_ctx=spectral_ctx)
```
Comment (nollety) [inline tests/kernel/test_gridvolume.py:15]: I assume what you removed here was not used

File: tests/test_presolver.py [hunk 2]
```diff
@@ -64,13 +64,14 @@ def test_resolve(presolver):
     presolver.clear()
 
     # We add a path
-    base = os.path.join(os.getenv("ERADIATE_DIR"), "eradiate")
+    base = os.getenv("ERADIATE_DIR")
     presolver.append(base)
+    print(presolver)
```
Comment (nollety) [inline tests/test_presolver.py:69]: Just out of curiosity, why is this print statement needed here?

**Review by nollety — APPROVED** [review]
@leroyvn Great, I like this layout, it makes a lot of sense in my opinion! I hardly think this will have any (negative) impact on my workflow.

File: tests/test_presolver.py (hunk 2 above)
Comment (leroyvn) [reply tests/test_presolver.py:69]: It's not, it's a mistake. Thanks for catching it!

File: tests/kernel/test_gridvolume.py (hunk 1 above)
Comment (leroyvn) [reply tests/kernel/test_gridvolume.py:15]: Indeed, I checked for usage and got no hit.

**Comment by leroyvn** [comment]
Seems like everything went well, let's merge!

**Pull request merged**